airflow-core/src/airflow/jobs/scheduler_job_runner.py

```diff
@@ -1769,6 +1770,25 @@ def _update_state(dag: DAG, dag_run: DagRun):
             _update_state(dag, dag_run)
             dag_run.notify_dagrun_state_changed()
 
+            if (deadline := dag.deadline) and isinstance(
+                deadline.reference, ReferenceModels.DagRunQueuedAtDeadline
+            ):
```

Comment:
Would it be a good idea to add a `get_dagrun_queued_at_deadline` on DAG like the creation deadline? Or remove the function for the creation deadline? It seems weird one has a function and one does not.

If this feels repetitive, maybe the function should be changed to like `get_deadline(kind: type[T]) -> T`

Reply:
I agree, I've removed it for now and if it ends up being repetitive, we could add a function to get deadline by reference type